src/learn/sfen_reader.h

```diff
@@ -220,29 +241,40 @@ namespace Learner{
             {
                 // Wait for the buffer to run out.
                 // This size() is read only, so you don't need to lock it.
-                while (!stop_flag && packed_sfens_pool.size() >= SFEN_READ_SIZE / THREAD_BUFFER_SIZE)
+                while (!stop_flag && packed_sfens_pool.size() >= sfen_read_size / thread_buffer_size)
                     sleep(100);
 
                 if (stop_flag)
                     return;
 
                 PSVector sfens;
-                sfens.reserve(SFEN_READ_SIZE);
+                sfens.reserve(sfen_read_size);
 
                 // Read from the file into the file buffer.
-                while (sfens.size() < SFEN_READ_SIZE)
+                while (sfens.size() < sfen_read_size)
                 {
                     std::optional<PackedSfenValue> p = sfen_input_stream->next();
                     if (p.has_value())
                     {
                         sfens.push_back(*p);
+                        ++numEntriesReadFromCurrentFile;
                     }
-                    else if(!open_next_file())
+                    else
                     {
-                        // There was no next file. Abort.
-                        std::cout << "..end of files." << std::endl;
-                        end_of_files = true;
-                        return;
+                        if (mode == SfenReaderMode::Cyclic
```

Comment:
I'm sorry if my understanding is wrong.  If there are files A, B and C, `Sequential` mode reads as "A -> B -> C", and `Cyclic` mode reads as "A -> A -> A -> ...".  Is this correct?  Do we need the mode to read as "A -> B -> C -> A -> B -> C -> ..."?
Also, `Cyclic` was not intuitive to me.  How about `RepeatOne`?  If we add the mode to read as "A -> B -> C -> A -> B -> C -> ...", `RepeatAll` would be good.

Reply:
> Could you please change the description in README.md? It still contains loop option.
I did change the readme for learn, gensfen still has the loop option.

> Cyclic mode reads as "A -> A -> A -> ...". Is this correct?
The cyclic mode works by appending each processed (read to the end) file that contained any data to the end of the queue (filenames). So when the user adds files `[A, B, C]` then after reading `A` is completed and it contained any data (wasn't empty) it's removed from the queue and added at the end, so the queue is then `[B, C, A]`. So it is `A -> B -> C -> A -> B -> C -> ...`.

`RepeatAll would be good.` IMO `RepeatAll` doesn't convey the indefinite nature of it, does it repeat once? or many times? `Cyclic` is consistent with the semantics of python's itertools `cycle`, that's why I thought it's well established in meaning.